### PR TITLE
[AWS Orgs] Update fetchers in each cycle to detect changes in accounts

### PR DESCRIPTION
**Pull request by orestisfl**
### Summary of your changes

This adds a new `Update()` method in the registry interface and an updater in the registry which allows the caller to update the fetchers in each cycle.

### Screenshot/Data

Tests from running cloudbeat locally (without an agent), adding and deleting an account without restarting cloudbeat.

**Account that was created at ~16:05:**
screenshot-2023-09-04T17:07:05 CEST
**Account that was deleted at ~16:15:**
screenshot-2023-09-04T17:07:15 CEST

The accounts also show up correctly in the dashboard, the one that is deleted shows up as "Last evaluated an hour ago"
image

### Related Issues

- Fixes: #1214 

### Checklist
- [X] I have added tests that prove my fix is effective or that my feature works
- [X] I have added the necessary README/documentation (if appropriate)

### CODE REVIEW

File: resources/fetching/registry/dynamic.go [hunk 1]
```diff
@@ -0,0 +1,107 @@
...
+	d.lock.RLock()
+	defer d.lock.RUnlock()
+	return d.registry.Run(ctx, key, metadata)
+}
+
+func (d *dynamic) Stop() {
+	d.lock.RLock()
+	defer d.lock.RUnlock()
+
+	d.registry.Stop()
```
Comment (amirbenun) [inline resources/fetching/registry/dynamic.go:82]: `dynamic.Stop` should also stop the scheduling from reoccurring

**Review by amirbenun — COMMENTED** [review]
Nice!

File: flavors/benchmark/aws_org.go [hunk 2]
```diff
@@ -58,18 +58,29 @@ func (a *AWSOrg) Initialize(ctx context.Context, log *logp.Logger, cfg *config.C
...
-	return registry.NewRegistry(
-			log,
-			factory.NewCisAwsOrganizationFactory(ctx, log, ch, accounts),
-		), cloud.NewDataProvider(
-			cloud.WithLogger(log),
-			cloud.WithAccount(*awsIdentity),
-		), cloud.NewIdProvider(), nil
+	cache := make(map[string]factory.FetchersMap)
+	reg := registry.NewRegistry(log, registry.WithUpdater(
+		func() (factory.FetchersMap, error) {
```
Comment (amirbenun) [inline flavors/benchmark/aws_org.go:63]: Nit: extract this function

File: resources/fetching/registry/registry_test.go [hunk 3]
```diff
@@ -256,3 +260,97 @@ func (c *boolFetcherCondition) Condition() bool {
 func (c *boolFetcherCondition) Name() string {
 	return c.name
 }
+
+func Test_registry_Update(t *testing.T) {
```
Comment (amirbenun) [inline resources/fetching/registry/registry_test.go:264]: Can you add another test where you initial the registry with no `Option`s?
The empty use case should probably just returns an error or hold an empty map that cannot be changed.
I want to make sure it doesn't panic.

**Review by amirbenun — APPROVED** [review]

File: resources/fetching/registry/registry_test.go (hunk 3 above)
Comment (orestisfl) [reply resources/fetching/registry/registry_test.go:264]: That would be pretty much equivalent to the `check nil` case

**Pull request merged**